Quantum-Control-Applications/Superconducting/Single Fixed Transmon/Use Case 3 - DRAG calibration/coarse_DRAG.py

```diff
@@ -0,0 +1,129 @@
+from qm.qua import *
+from configuration import *
+from qm.QuantumMachinesManager import QuantumMachinesManager
+import matplotlib.pyplot as plt
+import numpy as np
+from scipy import signal
+
+n_avg = 100
+
+cooldown_time = 5 * qubit_T1 // 4
+
+a_min = -1
+a_max = 1
+da = 0.05
+amps = np.arange(a_min, a_max + da / 2, da)  # + da/2 to add a_max to amplitudes
+
+iter_min = 0
+iter_max = 25
+d = 1
+iters = np.arange(iter_min, iter_max + 0.1, d)
+
+with program() as drag:
+    n = declare(int)
+    n_st = declare_stream()
+    a = declare(fixed)
+    I = declare(fixed)
+    Q = declare(fixed)
+    I_st = declare_stream()
+    Q_st = declare_stream()
+    state = declare(bool)
+    state_st = declare_stream()
+    it = declare(int)
+    pulses = declare(int)
+    I_g = declare(fixed)
+
+    with for_(n, 0, n < n_avg, n + 1):
+        # Notice it's + da/2 to include a_max (This is only for fixed!)
+        with for_(a, a_min, a < a_max + da / 2, a + da):
+            with for_(it, iter_min, it <= iter_max, it + d):
+                measure(
+                    "readout", "resonator", None, dual_demod.full("rotated_cos", "out1", "rotated_sin", "out2", I_g)
+                )
+                with while_(I_g > -0.0003):
+                    measure(
+                        "readout", "resonator", None, dual_demod.full("rotated_cos", "out1", "rotated_sin", "out2", I_g)
+                    )
```

Comment:
Is this here to wait a non deterministic time until the qubit is in the ground state ?
Since you seem to use single shot readout below, why don't you apply a pi pulse to bring the qubit faster to the ground state?

Reply:
I did not systematically tested that applying a pi pulse would get me better results so I just went with this